modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbStorage.java

```diff
@@ -462,7 +472,7 @@ private ScanCursor(RocksIterator iter, Predicate<SearchRow> filter) {
 
         /** {@inheritDoc} */
         @Override public boolean hasNext() {
-            while (isValid() && !filter.test(new SimpleDataRow(iter.key(), null)))
+            while (isValid() && !filter.test(new SimpleDataRow(iter.key(), iter.value())))
```

Comment:
Is there a way to avoid "iter.value()" here? It's never used in the filter

Reply:
why is it bad? There's a way, but we would need to introduce an implementation of `SearchRow`

Reply:
I couldn't find how exactly iterator reads the value. It might involve unnecessary IO